password/impl/src/main/java/org/wildfly/security/password/impl/MaskedPasswordImpl.java

```diff
@@ -64,24 +62,38 @@
     private final int iterationCount;
     private final byte[] salt;
     private final byte[] maskedPasswordBytes;
+    private final byte[] initializationVector;
 
-    private MaskedPasswordImpl(final String algorithm, final char[] initialKeyMaterial, final int iterationCount, final byte[] salt, final byte[] maskedPasswordBytes, final boolean validated) throws InvalidKeySpecException {
+
+    private MaskedPasswordImpl(final String algorithm, final char[] initialKeyMaterial, final int iterationCount, final byte[] salt, final byte[] initializationVector, final byte[] maskedPasswordBytes) throws InvalidKeySpecException {
         Assert.checkMinimumParameter("iterationCount", 1, iterationCount);
-        // perform an unmask to validate parameters
-        if (! validated) unmask(algorithm, initialKeyMaterial, iterationCount, salt, maskedPasswordBytes);
         this.algorithm = algorithm;
         this.initialKeyMaterial = initialKeyMaterial;
         this.iterationCount = iterationCount;
         this.salt = salt;
         this.maskedPasswordBytes = maskedPasswordBytes;
+        this.initializationVector = initializationVector;
+        unmask();
```

Comment:
I think it would be good to add the original comment on this line: ```// perform an unmask to validate parameters```

Reply:
Fixed